more/pixels/pixelMintController.js

```diff
@@ -55,28 +61,38 @@ export function makeMintController(assay) {
     rights = makePrivateName(); // reset rights
   }
 
-  function recordPayment(src, payment, amount, srcNewRightsAmount) {
-    rights.set(src, srcNewRightsAmount);
-    setLocation(srcNewRightsAmount, src);
-    rights.init(payment, amount);
-    setLocation(amount, payment);
+  // creating a purse creates a new mapping from the purse to the
+  // amount within the purse
+  function recordNewPurse(purse, initialAmount) {
+    rights.init(purse, initialAmount);
+    setPixelsToAsset(initialAmount, purse);
   }
 
-  function recordDeposit(
-    srcPayment,
-    srcNewRightsAmount,
-    purse,
-    purseNewRightsAmount,
+  // creating a payment creates a new mapping from the payment to the
+  // amount within the payment
+  // It also takes that amount from the source purseOrPayment, so
+  // the source must be updated with the new (lesser) newPurseOrPaymentAmount
+  function recordNewPayment(
+    srcPurseOrPayment,
+    payment,
+    paymentAmount,
+    newSrcAmount,
   ) {
-    rights.set(srcPayment, srcNewRightsAmount);
-    setLocation(srcNewRightsAmount, srcPayment);
-    rights.set(purse, purseNewRightsAmount);
-    setLocation(purseNewRightsAmount, purse);
+    rights.init(payment, paymentAmount);
+    rights.set(srcPurseOrPayment, newSrcAmount);
+
+    setPixelsToAsset(paymentAmount, payment);
+    setPixelsToAsset(newSrcAmount, srcPurseOrPayment);
   }
 
-  function recordMint(purse, initialAmount) {
-    rights.init(purse, initialAmount);
-    setLocation(initialAmount, purse);
+  // a deposit (putting a payment or part of a payment into a purse)
+  // changes the amounts of both the purse and the payment
+  function recordDeposit(payment, newPaymentAmount, purse, newPurseAmount) {
```

Comment:
I can imagine that we'd do some extra record keeping or balance checking to take advantage of the distinctions between recordNewPurse, recordNewPayment, and recordDeposit, but we're not doing any of that. It seems like the interface would be simpler if we just had two functions: 
updateBalance(purseOrPayment, newBalance)
and 
recordNewAsset(purseOrPayment, initialBalance).

Otherwise, it doesn't seem like the distinctions between the three current functions is gaining us much.

Reply:
Hmm,  creating a new Payment and creating a new Purse does different things, though. Were you imagining that this logic would get pushed to issuer.js and mintController would be very simple?

Reply:
I'm trying this out, and I think I like it